backend/src/schema/resolvers/cartResolvers.ts

```diff
@@ -0,0 +1,341 @@
+
+import { GraphQLError } from "graphql";
+import MenuItem,{IMenuItem} from "../../models/MenuItem";
+import Cart,{ICart} from "../../models/Cart";
+import logger from "../../utils/logger";
+import mongoose from "mongoose";
+import { AddToCartInput, MyContext, RemoveItemInput,DecreaseQuantityInput } from "../types/types";
+
+interface PopulatedCart extends Omit<ICart, "items"> {
+    items: {
+      menuItemId: IMenuItem; // Fully populated MenuItem
+      quantity: number;
+    }[];
+  }
+
+export const cartResolvers = {
+    Query:{
+        async getAllCartItems  (_: any, __:any, context: MyContext)  {
+            try {
+              // Ensure the user is authenticated
+              if (!context.user) {
+                throw new GraphQLError("Authentication required", {
+                  extensions: { code: "UNAUTHORIZED" },
+                });
+              }
+          
+            //   if (!context.user.id ) {
+            //     throw new GraphQLError("Access denied", {
+            //       extensions: { code: "FORBIDDEN" },
+            //     });
+            //   }
+          
+              // Fetch the user's cart and populate menuItemId
+              let cart : PopulatedCart|null  = await Cart.findOne({ userId:context.user.id  }).populate({
+                path: "items.menuItemId",
+                model: MenuItem,
+              }).lean<PopulatedCart | null>();
+          
+              if (!cart) {
+                return null; // Return null if the cart doesn't exist
+              }
+            //   console.log({...cart.items})
+          
+              // Map data to match the GraphQL type definition
+              const formattedCart  = {
+                id: cart._id,
+                userId: cart.userId,
+                items: cart.items.map((item) => ({
+                  menuItem: {
+                    id: item.menuItemId._id,
+                    name: item.menuItemId.name,
+                    description: item.menuItemId.description,
+                    image: item.menuItemId.image,
+                    price: item.menuItemId.price,
+                    isVeg: item.menuItemId.isVeg,
+                    categoryId: item.menuItemId.categoryId,
+                    isActive: item.menuItemId.isActive,
+                  },
+                  quantity: item.quantity,
+                })),
+              };
+          
+              logger.info(`Fetched cart for user ${context.user.id}`);
+
+              return formattedCart;
```

Comment:
does the condition added here  and reformatting formatted object makes sense?
```suggestion

logger.info(`Fetched cart for user ${context.user.id}`);

return cart;
```